UndertaleModLib/Util/TextureWorker.cs

```diff
@@ -1,217 +1,250 @@
+        public static IMagickImage<byte> ResizeImage(IMagickImage<byte> image, int width, int height)
         {
-            Bitmap bm = (Bitmap)_imageConverter.ConvertFrom(byteArray);
-
-            if (bm != null && (bm.HorizontalResolution != (int)bm.HorizontalResolution ||
-                               bm.VerticalResolution != (int)bm.VerticalResolution))
+            if (image.Width == width && image.Height == height)
             {
-                // Correct a strange glitch that has been observed in the test program when converting 
-                //  from a PNG file image created by CopyImageToByteArray() - the dpi value "drifts" 
-                //  slightly away from the nominal integer value
-                bm.SetResolution((int)(bm.HorizontalResolution + 0.5f),
-                                 (int)(bm.VerticalResolution + 0.5f));
+                // We already have the correct dimensions, so just make a clone of the image.
+                // If we don't clone here, we have potential ownership/disposal issues.
+                return image.Clone();
             }
 
-            return bm;
+            // Clone and resize using bilinear interpolation
+            IMagickImage<byte> newImage = image.Clone();
+            newImage.InterpolativeResize(width, height, PixelInterpolateMethod.Bilinear);
+            return newImage;
         }
 
-        // This should perform a high quality resize.
-        // Grabbed from https://stackoverflow.com/questions/1922040/how-to-resize-an-image-c-sharp
-        public static Bitmap ResizeImage(Image image, int width, int height)
+        /// <summary>
+        /// Reads collision mask data from the given file path, and required width/height.
+        /// </summary>
+        /// <exception cref="Exception">If the loaded image dimensions do not match the required width/height</exception>
+        public static byte[] ReadMaskData(string filePath, int requiredWidth, int requiredHeight)
         {
-            var destRect = new Rectangle(0, 0, width, height);
-            var destImage = new Bitmap(width, height);
-
-            destImage.SetResolution(image.HorizontalResolution, image.VerticalResolution);
-
-            using (var graphics = Graphics.FromImage(destImage))
+            List<byte> bytes;
+            using (MagickImage image = ReadBGRAImageFromFile(filePath))
             {
-                graphics.CompositingMode = CompositingMode.SourceCopy;
-                graphics.CompositingQuality = CompositingQuality.HighQuality;
-                graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
-                graphics.SmoothingMode = SmoothingMode.HighQuality;
-                graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
-
-                using (var wrapMode = new ImageAttributes())
+                // Verify width/height match required width/height
+                if (image.Width != requiredWidth || image.Height != requiredHeight)
```

Comment:
so, why do we need the custom width/height as parameters if we always end up using the image's one?

Reply:
bump